### PR TITLE
Add some improvements to the module, add missing LICENSE, update docs

### PR DESCRIPTION
**Pull request by gstlt**
I believe these changes will make the module easier to use. Please review and let me know how this looks for you. I'll be happy to do any needed amends.

Changes as follows:
- removed provider - not desired in a module
- changed rds_instance_name to more accurate rds_instance_identifier (to be consistent with Terraform nomenclature)
- create own parameter group so we do not need to pass another ID to the module
- get subnets in a list format for a case where we want to choose more than 2 subnets
- create own security group and connect it to provided VPC
- add description to variables
- add LICENSE
- update README.md

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -1,63 +1,85 @@
...
+    description = "Instance sizing"
+    # Valid values
+    # https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/Concepts.DBInstanceClass.html
+}
 
 variable "database_name" {
     description = "The name of the database to create"
 }
 
+# Sefl-explainatory variables
```
Comment (antonbabenko) [inline variables.tf]: "Self"

File: variables.tf [hunk 2]
```diff
@@ -1,63 +1,85 @@
...
+# We're "cloning" default ones, but we need to specify which should be copied
 variable "db_parameter_group" {
-    default = "default.mysql5.6"
+    # default = "mysql5.6"
+    default = "postgres9.5"
 }
 
 variable "publicly_accessible" {
+    description = "If database can be publicly available (NOT recommended)"
     default = "false"
```
Comment (antonbabenko) [inline variables.tf]: false can be without quotes

File: variables.tf [hunk 3]
```diff
@@ -1,63 +1,85 @@
...
-    // --engine-version
+    # For valid engine versions, see:
+    # See http://docs.aws.amazon.com/cli/latest/reference/rds/create-db-instance.html
+    # --engine-version
 
 }
 
-variable "rds_instance_class" {}
+variable "rds_instance_class" {
+    description = "Instance sizing"
```
Comment (antonbabenko) [inline variables.tf]: I think this is more correct - "Class of RDS instance"

File: variables.tf [hunk 4]
```diff
@@ -1,63 +1,85 @@
...
+    # - sqlserver-*
+    # See http://docs.aws.amazon.com/cli/latest/reference/rds/create-db-instance.html
+    # --engine
 }
 
 variable "rds_engine_version" {
-    // For valid engine versions, see:
-    // See http://docs.aws.amazon.com/cli/latest/reference/rds/create-db-instance.html
-    // --engine-version
+    # For valid engine versions, see:
```
Comment (antonbabenko) [inline variables.tf]: Add `description` to all variables, please.

File: main.tf [hunk 5]
```diff
@@ -7,35 +7,68 @@
...
 resource "aws_db_subnet_group" "main_db_subnet_group" {
-    name = "${var.rds_instance_name}-subnetgrp"
+    name = "${var.rds_instance_identifier}-subnetgrp"
     description = "RDS subnet group"
-    subnet_ids = ["${var.subnet_az1}", "${var.subnet_az2}"]
+    subnet_ids = ["${var.subnets}"]
+}
+
+# Security groups
+resource "aws_security_group" "main_db_access" {
```
Comment (antonbabenko) [inline main.tf]: Split this in `aws_security_group` and several `aws_security_group_rule`. Also add to output `${aws_security_group. main_db_access.id}`

Comment (gstlt) [reply main.tf]: Thanks! I'm on it.
Additionally I've spotted that db_parameter_group has set a default which might be not desired if we're not using assumed DB. I'll update it as well shortly.

**Pull request merged**